### PR TITLE
WIP Colaboration branch

### PR DESCRIPTION
**Pull request by rsoeldner**

### CODE REVIEW

File: src/main/scala/gecko/FrameIndex.scala [hunk 1]
```diff
@@ -2,58 +2,102 @@ package gecko
...
 
-final case class FrameIndex[@specialized(Int, Double, Boolean, Long) A: ClassTag](
-    underlying: Array[A],
-    private[gecko] val indexes: Array[Int]
-) {
-  @inline def apply(i: Int): A = underlying(i)
+final case class FrameIndex[@specialized(Int, Double, Boolean, Long) A: ClassTag](underlying: Array[A]) {
 
-  @inline def index(i: Int): Int = indexes(i)
+  @inline def apply(i: Int): A = underlying(i)
```
Comment (jmcardon) [inline src/main/scala/gecko/FrameIndex.scala]: we should consider having this private,  bounds checked, or both.

File: src/main/scala/gecko/FrameIndex.scala [hunk 2]
```diff
@@ -2,58 +2,102 @@ package gecko
...
   }
 
-  /** Remove elements at index i
+  /** Alias for ++
     *
+    * @param other
+    * @return
     */
-  def unsafeRemoveIx(i: Int): FrameIndex[A] = FrameIndex(removeElemAt(underlying, i), removeElemAt(indexes, i))
+  def concat(other: FrameIndex[A]) = this ++ other
```
Comment (jmcardon) [inline src/main/scala/gecko/FrameIndex.scala]: :+1:

File: src/main/scala/gecko/FrameIndex.scala [hunk 3]
```diff
@@ -2,58 +2,102 @@ package gecko
...
 object FrameIndex {
-  def default(size: Int): FrameIndex[Int] = {
-    val ix = Array.range(0, size)
-    FrameIndex(ix, ix)
-  }
-  def empty[A: ClassTag]: FrameIndex[A] =
-    FrameIndex(Array.empty[A], Array.empty[Int])
+  def default(size: Int): FrameIndex[Int] = FrameIndex(Array.range(0, size))
+
+  def fromSeq[C: ClassTag](c: Seq[C]): FrameIndex[C] = FrameIndex[C](c.toArray)
```
Comment (jmcardon) [inline src/main/scala/gecko/FrameIndex.scala]: I wonder if we can @specialized C, for better performance. 

Maybe it's not necessary but checking would be nice.

File: src/main/scala/gecko/FrameIndex.scala (hunk 1 above)
Comment (rsoeldner) [reply src/main/scala/gecko/FrameIndex.scala]: :+1:

**Pull request merged**